[Discovery] Query execution fails with RuntimeException if query string contains '|' character

The following code using Discovery service Java Client API fails with error.
If I change the query to just "author", it works fine. 

```
import com.ibm.watson.developer_cloud.discovery.v1.Discovery;
import com.ibm.watson.developer_cloud.discovery.v1.model.query.QueryRequest;
import com.ibm.watson.developer_cloud.discovery.v1.model.query.QueryResponse;
import com.ibm.watson.developer_cloud.http.ServiceCall;

public class DiscoveryTest {
    static private String endPoint = "https://gateway.watsonplatform.net/discovery/api";
    static private String username = "[User name]";
    static private String password = "[Password]";
    static private String environmentId = "[Environment ID]";
    static private String collectionId = "[Collection ID]";

//    static private String query = "author";
    static private String query = "author|writer";
    
    static public void main(String args[]){
        Discovery service = new Discovery("2016-12-11");
        service.setEndPoint(endPoint);
        service.setUsernameAndPassword(username, password);
        
        QueryRequest.Builder builder = new QueryRequest.Builder(environmentId, collectionId);
        QueryRequest queryRequest = builder.query(query).build();
        
        ServiceCall<QueryResponse> call = service.query(queryRequest);
        QueryResponse response = call.execute();
        
        System.out.println(response.toString());
    }
}

```
- Expected behavior
The query is successfully executed and query result is responded.
- Actual behavior
Runtime exception
```
Exception in thread "main" java.lang.RuntimeException: Error reading the http response
	at com.ibm.watson.developer_cloud.util.ResponseUtils.getString(ResponseUtils.java:110)
	at com.ibm.watson.developer_cloud.service.WatsonService.getErrorMessage(WatsonService.java:290)
	at com.ibm.watson.developer_cloud.service.WatsonService.processServiceCall(WatsonService.java:410)
	at com.ibm.watson.developer_cloud.service.WatsonService$1.execute(WatsonService.java:180)
	at DiscoveryTest.main(DiscoveryTest.java:24)
Caused by: java.io.EOFException
	at okio.RealBufferedSource.require(RealBufferedSource.java:59)
	at okio.GzipSource.consumeHeader(GzipSource.java:114)
	at okio.GzipSource.read(GzipSource.java:73)
	at okio.RealBufferedSource.request(RealBufferedSource.java:66)
	at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:387)
	at okio.RealBufferedSource.rangeEquals(RealBufferedSource.java:371)
	at okhttp3.internal.Util.bomAwareCharset(Util.java:412)
	at okhttp3.ResponseBody.string(ResponseBody.java:173)
	at com.ibm.watson.developer_cloud.util.ResponseUtils.getString(ResponseUtils.java:107)
	... 4 more
```  
- JDK version
JDK 1.8.0
- java-sdk version
3.7.0

--- a/core/src/main/java/com/ibm/watson/developer_cloud/http/RequestBuilder.java
+++ b/core/src/main/java/com/ibm/watson/developer_cloud/http/RequestBuilder.java
@@ -209,7 +209,7 @@ public class RequestBuilder {
     for (final NameValue param : queryParams) {
       builder.addQueryParameter(param.getName(), param.getValue());
     }
-    return builder.build().url().toString();
+    return builder.build().uri().toString();
   }
 
   /**

--- a/discovery/src/test/java/com/ibm/watson/developer_cloud/discovery/v1/DiscoveryServiceIT.java
+++ b/discovery/src/test/java/com/ibm/watson/developer_cloud/discovery/v1/DiscoveryServiceIT.java
@@ -338,6 +338,24 @@ public class DiscoveryServiceIT extends WatsonServiceTest {
   }
 
   @Test
+  public void getConfigurationsWithFunkyNameIsSuccessful() {
+    String uniqueConfigName = uniqueName + " with \"funky\" ?x=y&foo=bar ,[x](y) ~!@#$%^&*()-+ {} | ;:<>\\/ chars";
+
+    CreateConfigurationRequest.Builder createBuilder = new CreateConfigurationRequest.Builder(environmentId);
+    Configuration configuration = getTestConfiguration(DISCOVERY_TEST_CONFIG_FILE);
+    configuration.setName(uniqueConfigName);
+    createBuilder.configuration(configuration);
+    CreateConfigurationResponse createResponse = createConfiguration(createBuilder.build());
+
+    GetConfigurationsRequest.Builder getBuilder = new GetConfigurationsRequest.Builder(environmentId);
+    getBuilder.name(uniqueConfigName);
+    GetConfigurationsResponse getResponse = discovery.getConfigurations(getBuilder.build()).execute();
+
+    assertEquals(1, getResponse.getConfigurations().size());
+    assertEquals(uniqueConfigName, getResponse.getConfigurations().get(0).getName());
+  }
+
+  @Test
   public void updateConfigurationIsSuccessful() {
     CreateConfigurationResponse createResponse = createTestConfig();
 
@@ -802,6 +820,17 @@ public class DiscoveryServiceIT extends WatsonServiceTest {
       Operation operation = normalization.getOperation();
       assertEquals(true, Arrays.asList(operations).contains(operation.name()));
     }
+  }
+
+  @Test
+  public void issueNumber654() {
+    String collectionId = setupTestDocuments();
+    QueryRequest.Builder queryBuilder = new QueryRequest.Builder(environmentId, collectionId);
+    queryBuilder.query("field:1|3");
+    QueryResponse queryResponse = discovery.query(queryBuilder.build()).execute();
+
+    assertEquals(new Long(2), queryResponse.getMatchingResults());
+    assertEquals(2, queryResponse.getResults().size());
   }
 
   private CreateEnvironmentResponse createEnvironment(CreateEnvironmentRequest createRequest) {

